test/node-api/test_buffer/test_buffer.c

```diff
@@ -123,6 +125,39 @@ static napi_value invalidObjectAsBuffer(napi_env env, napi_callback_info info) {
   return notTheBuffer;
 }
 
+static void jsCallback(napi_env env, void* finalize_data, void* finalize_hint) {
+  (void)finalize_hint;
+  free(finalize_data);
+  deleterCallCount++;
+}
+
+static void finalizeBuffer(const napi_env* env,
+                           void* data,
+                           void* finalize_hint) {
+  node_api_post_finalizer(env, data, finalize_hint, jsCallback);
+}
+
+static napi_value bufferFromArrayBuffer(napi_env env, napi_callback_info info) {
+  napi_status status;
+  napi_value arraybuffer;
+  void* data;
+  napi_value buffer;
+  size_t byte_length = 1024;
+  size_t byte_offset = 0;
+
+  status = napi_create_arraybuffer(env, byte_length, &data, &arraybuffer);
+  NODE_API_ASSERT(env, status == napi_ok, "Failed to create arraybuffer");
+
+  status = node_api_create_buffer_from_arraybuffer(
+      env, arraybuffer, byte_offset, byte_length, &buffer);
+  NODE_API_ASSERT(
+      env, status == napi_ok, "Failed to create buffer from arraybuffer");
+
+  napi_add_finalizer(env, buffer, data, finalizeBuffer, NULL, NULL);
```

Comment:
Is this finalizer necessary?

Reply:
I added this at the suggestion of @vmoroz

Reply:
The suggestion https://github.com/nodejs/node/pull/54505#issuecomment-2354355409 is that the finalizer in https://nodejs.org/docs/latest/api/n-api.html#napi_create_external_arraybuffer will be more eager to be invoked and in this finalizer it can not run JavaScript codes. `node_api_post_finalizer` can help with invoking JS code in a finalizer.

However, I don't think we need finalizer at all in this PR. https://nodejs.org/docs/latest/api/n-api.html#napi_create_arraybuffer does not require a finalizer. `napi_create_buffer_from_arraybuffer` can directly create a Buffer instance with this ArrayBuffer.

Reply:
I followed what you said thank you @legendecas